### PR TITLE
Replace reply data tuple with struct

### PR DESCRIPTION
**Pull request by breathx**
`Message`s reply now contains special named struct instead of just tuple `(MessageId, ExitCode)`. This increases readability of events in scanners like polkadot js apps 

@gear-tech/dev

### CODE REVIEW

File: core/src/message/mod.rs [hunk 1]
```diff
@@ -92,7 +92,7 @@ impl DispatchKind {
 
 /// Message packet.
 ///
-/// Provides common behaviour for any message's packet: accessing to payload, gas limit and value.
+/// Provides common behavior for any message's packet: accessing to payload, gas limit and value.
```
Comment (shamilsan) [inline core/src/message/mod.rs:95]: You don't like British English 😄

File: core-backend/sandbox/src/funcs.rs [hunk 2]
```diff
@@ -360,15 +360,16 @@ where
...
-            .with_fallible(|ext| ext.reply_to().map_err(FuncError::Core))
+            .with_fallible(|ext| ext.reply_details().map_err(FuncError::Core))
             .map_err(|e| {
                 ctx.err = e;
                 HostError
             })?;
 
-        if let Some((_, exit_code)) = reply_tuple {
+        if let Some(details) = opt_details {
+            let exit_code = details.into_parts().1;
```
Comment (shamilsan) [inline core-backend/sandbox/src/funcs.rs:372]: Why not just
```suggestion
            let exit_code = details.exit_code();
```

File: core-backend/sandbox/src/funcs.rs [hunk 3]
```diff
@@ -595,15 +596,16 @@ where
...
-            .with_fallible(|ext| ext.reply_to().map_err(FuncError::Core))
+            .with_fallible(|ext| ext.reply_details().map_err(FuncError::Core))
             .map_err(|err| {
                 ctx.err = err;
                 HostError
             })?;
 
-        if let Some((message_id, _)) = maybe_message_id {
+        if let Some(details) = opt_details {
+            let message_id = details.into_parts().0;
```
Comment (shamilsan) [inline core-backend/sandbox/src/funcs.rs:608]: Also, do we really need `into_parts` as parts are copy types?
```suggestion
            let message_id = details.reply_to();
```

**Review by shamilsan — APPROVED** [review]
I like using structs instead of tuples in such kind of circumstances 👍

File: core-backend/sandbox/src/funcs.rs (hunk 3 above)
Comment (breathx) [reply core-backend/sandbox/src/funcs.rs:608]: I don't see the reason to clone/copy even copy-types when there is no need to access details further. Move-semantic makes sense for me even in such small things.
Isn't this destruction cheaper then copy?

Comment (breathx) [reply core-backend/sandbox/src/funcs.rs:608]: We can provide something like
`fn into_reply_to(self) -> MessageId` for comfortable usage

**Review by osipov-mit — APPROVED** [review]

**Pull request merged**